src/System/Hooks/usePrefetchRoute.tsx

```diff
@@ -0,0 +1,75 @@
+import { fetchQuery } from "react-relay"
+import { OperationType, Subscription } from "relay-runtime"
+import { useRouter } from "System/Hooks/useRouter"
+import { useSystemContext } from "System/Hooks/useSystemContext"
+import { findRoutesByPath } from "System/Router/Utils/routeUtils"
+
+export const usePrefetchRoute = (
+  initialPath?: string
+): { prefetch: (path?: string) => Array<Subscription | null> | null } => {
+  const { relayEnvironment } = useSystemContext()
+
+  const { match } = useRouter()
+
+  const prefetch = (path = initialPath) => {
+    if (!path) {
+      return null
+    }
+
+    const routes = findRoutesByPath({ path })
+
+    const querySubscriptions = routes.map(foundRoute => {
+      if (!foundRoute || !foundRoute.route) {
+        return null
+      }
+
+      const {
+        match: { params },
+      } = foundRoute
+
+      const {
+        route: { query, prepareVariables },
+      } = foundRoute
+
+      const variables = (() => {
+        if (!prepareVariables) {
+          return params
+        }
+
+        return prepareVariables(params, match) as OperationType["variables"]
+      })()
+
+      const isPrefetchable = !!(query && variables)
+
+      if (!isPrefetchable) {
+        return null
+      }
+
+      const subscription = fetchQuery(relayEnvironment, query, variables, {
+        fetchPolicy: "store-or-network",
```

Comment:
So here we're using store-**or**-network instead of store-**and**-network because we don't want to refetch an already prefetched route, right?